### PR TITLE
[REVIEW] Add interfaces for authorization server

### PR DESCRIPTION
**Pull request by RobKamstraMD**
@BauRo 
@JonnieMD

### CODE REVIEW

File: src/AuthorizationServer.php [hunk 1]
```diff
@@ -111,6 +112,24 @@ public function addGrantType(GrantTypeInterface $grantType, $identifier = null)
     }
 
     /**
+     * Return a grant type class
+     *
+     * @param string $grantType The grant type identifier
+     *
+     * @return GrantTypeInterface
+     *
+     * @throws
```
Comment (BauRo) [inline src/AuthorizationServer.php]: can you tell what it Throws?

File: src/AuthorizationServer.php [hunk 2]
```diff
@@ -111,6 +112,24 @@ public function addGrantType(GrantTypeInterface $grantType, $identifier = null)
...
+     */
+    public function getGrantType($grantType)
+    {
+        if (isset($this->grantTypes[$grantType])) {
+            return $this->grantTypes[$grantType];
+        }
+
+        throw new Exception\InvalidGrantException($grantType);
```
Comment (BauRo) [inline src/AuthorizationServer.php]: Can you add this to use statement?

File: src/Grant/AbstractGrant.php [hunk 3]
```diff
@@ -45,7 +45,7 @@
     /**
      * AuthServer instance
      *
-     * @var \League\OAuth2\Server\AuthorizationServer
+     * @var \League\OAuth2\Server\ServerInterface\AuthorizationServer
```
Comment (BauRo) [inline src/Grant/AbstractGrant.php]: only AuthorizationServer is enough here. :-)

File: src/ServerInterface/GrantServer.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public function hasGrantType($identifier);
+
+    /**
+     * Return a grant type class
+     *
+     * @param string $grantType The grant type identifier
+     *
+     * @return GrantTypeInterface
+     *
+     * @throws
```
Comment (BauRo) [inline src/ServerInterface/GrantServer.php]: Can you tell what it throws?

File: src/AuthorizationServer.php (hunk 1 above)
Comment (RobKamstraMD) [inline src/AuthorizationServer.php]: nope

File: src/Grant/AbstractGrant.php (hunk 3 above)
Comment (RobKamstraMD) [inline src/Grant/AbstractGrant.php]: we changed this concept

File: src/ServerInterface/GrantServer.php (hunk 4 above)
Comment (RobKamstraMD) [inline src/ServerInterface/GrantServer.php]: nope

**Comment by BauRo** [comment]
:+1:

**Pull request merged**